src/test/java/com/ibm/cloud/sdk/core/test/security/Cp4dAuthenticatorTest.java

```diff
@@ -281,14 +281,37 @@ public void testApiErrorBadRequest() throws Throwable {
 
     Request.Builder requestBuilder = new Request.Builder().url("https://test.com");
 
+    // Calling authenticate should result in an exception.
+    try {
+      authenticator.authenticate(requestBuilder);
+      fail("Expected authenticate() to result in exception!");
+    } catch (RuntimeException excp) {
+      assertTrue(excp instanceof ServiceResponseException);
+    } catch (Throwable t) {
+      fail("Expected RuntimeException, not " + t.getClass().getSimpleName());
+    }
+  }
+  
+  @Test
+  public void testApiErrorResponse() throws Throwable {
+    server.enqueue(jsonResponse("{'}"));
+
+    // Mock current time to ensure the token is valid.
+    PowerMockito.mockStatic(Clock.class);
+    PowerMockito.when(Clock.getCurrentTimeInSeconds()).thenReturn((long) 100);
+
+    CloudPakForDataAuthenticator authenticator = new CloudPakForDataAuthenticator(url, testUsername, testPassword);
+
+    Request.Builder requestBuilder = new Request.Builder().url("https://test.com");
+
     // Calling authenticate should result in an exception.
     try {
       authenticator.authenticate(requestBuilder);
       fail("Expected authenticate() to result in exception!");
     } catch (RuntimeException excp) {
       Throwable causedBy = excp.getCause();
       assertNotNull(causedBy);
-      assertTrue(causedBy instanceof ServiceResponseException);
+      assertFalse(causedBy instanceof ServiceResponseException);
```

Comment:
I think it would be better to check for the specific exception type that we'd expect in this "bad json response" scenario (i.e. a JsonParseException), rather than just make sure it's not a ServiceResponseException.